rotkehlchen/chain/evm/decoding/metamask/decoder.py

```diff
@@ -128,12 +125,16 @@ def _decode_swap(self, context: DecoderContext) -> DecodingOutput:
             else:
                 fee_asset = out_event.asset.resolve_to_evm_token()
 
-        # if fee_asset is a token, update only if it's in the Receive event
+        # update the in_event/out_event to adjust their balance with fees
         fee_amount = asset_normalized_value(amount=fee_raw, asset=fee_asset)
-        full_amount = to_amount + fee_amount
         if in_event.asset == fee_asset:
+            full_amount = in_event.balance.amount + fee_amount
```

Comment:
yes I agree with @yabirgb

In both cases you can also save one line by not making a temporary variable.

```python
in_event.balance.amount += fee_amount
in_event.notes = f'Receive {in_event.balance.amount} {fee_asset.symbol} as the result of a metamask swap' # noqa: E501
```

Do same logic for the out event